pkg/benchmarkjunit/parse.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// reSuiteStart identifies the start of a new TestSuite and captures the package path.
+	// Matches lines like "pkg: k8s.io/test-infra/experiment/dummybenchmarks"
+	reSuiteStart = regexp.MustCompile(`^pkg:\s+(\S+)\s*$`)
+	// reSuiteEnd identifies the end of a TestSuite and captures the overall result, package path, and runtime.
+	// Matches lines like:
+	// "ok  	k8s.io/test-infra/experiment/dummybenchmarks/subpkg	1.490s"
+	// "FAIL	k8s.io/test-infra/experiment/dummybenchmarks	17.829s"
+	reSuiteEnd = regexp.MustCompile(`^(ok|FAIL)\s+(\S+)\s+(\S+)\s*$`)
+	// reBenchMetrics identifies lines with metrics for successful Benchmarks and captures the name, op count, and metric values.
+	// Matches lines like:
+	// "Benchmark-4                 	20000000	       77.9 ns/op"
+	// "BenchmarkAllocsAndBytes-4   	10000000	       131 ns/op	 152.50 MB/s	     112 B/op	       2 allocs/op"
+	reBenchMetrics = regexp.MustCompile(`^(Benchmark\S*)\s+(\d+)\s+([\d\.]+) ns/op(?:\s+([\d\.]+) MB/s)?(?:\s+([\d\.]+) B/op)?(?:\s+([\d\.]+) allocs/op)?\s*$`)
+	// reActionLine identifies lines that start with "--- " and denote the start of log output and/or a skipped or failed Benchmark.
+	// Matches lines like:
+	// "--- BENCH: BenchmarkLog-4"
+	// "--- SKIP: BenchmarkSkip"
+	// "--- FAIL: BenchmarkFatal"
+	reActionLine = regexp.MustCompile(`^--- (BENCH|SKIP|FAIL):\s+(\S+)\s*$`)
+)
+
+// Property defines the xml element that stores additional metrics about each benchmark.
+type Property struct {
+	Name  string `xml:"name,attr"`
+	Value string `xml:"value,attr"`
+}
+
+// Properties defines the xml element that stores the list of properties that are associated with one benchmark.
+type Properties struct {
+	PropertyList []Property `xml:"property"`
+}
+
+// Failure defines the xml element that stores failure text.
+type Failure struct {
+	Text string `xml:",chardata"`
+}
+
+// TestCase defines the xml element that stores all information associated with one benchmark.
+type TestCase struct {
```

Comment:
We don't want to use/extend https://github.com/kubernetes/test-infra/blob/master/testgrid/metadata/junit/junit.go?